homeassistant/components/anglian_water/config_flow.py

```diff
@@ -0,0 +1,86 @@
+"""Implement a config flow for anglian_water."""
+
+from __future__ import annotations
+
+from pyanglianwater import API
+from pyanglianwater.exceptions import (
+    InvalidPasswordError,
+    InvalidUsernameError,
+    ServiceUnavailableError,
+)
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers import selector
+
+from .const import CONF_DEVICE_ID, DOMAIN, LOGGER
+
+
+class BlueprintFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Blueprint."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self,
+        user_input: dict | None = None,
+    ) -> config_entries.ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        _errors = {}
+        if user_input is not None:
+            try:
+                if user_input.get(CONF_DEVICE_ID, "") == "":
+                    auth = await API.create_via_login(
+                        email=user_input[CONF_USERNAME],
+                        password=user_input[CONF_PASSWORD],
+                    )
+                else:
+                    auth = await API.create_via_login_existing_device(
+                        email=user_input[CONF_USERNAME],
+                        password=user_input[CONF_PASSWORD],
+                        dev_id=user_input[CONF_DEVICE_ID],
+                    )
+            except InvalidUsernameError as exception:
+                LOGGER.warning(exception)
+                _errors["base"] = "auth"
+            except InvalidPasswordError as exception:
+                LOGGER.warning(exception)
+                _errors["base"] = "auth"
+            except ServiceUnavailableError:
+                LOGGER.warning(
+                    "Anglian Water app service is unavailable. Check the app for more information"
+                )
+                _errors["base"] = "maintenance"
+            else:
+                user_input[CONF_DEVICE_ID] = auth.device_id
+                return self.async_create_entry(
+                    title=user_input[CONF_USERNAME],
+                    data=user_input,
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_DEVICE_ID,
+                        default=(user_input or {}).get(CONF_DEVICE_ID, ""),
+                    ): selector.TextSelector(),
```

Comment:
Can you elaborate on what a device id is?

Reply:
device id is a unique identifier that identifies the connecting device on anglian water's side, usually its a form of the device fingerprint that runs the android app. Leaving this alone will auto generate one in the background.

Its needed to retrieve water data.

Reply:
So why does someone needs to set it?

Reply:
I have removed this, but it is still required otherwise we might overload AW's API with fake device registration requests.

Reply:
Would it help if you remove the device if the user removes the integration?

Reply:
Sorry, not sure I'm following..

The device doesn't get unregistered as there are no endpoints for that, only registration. Each time a new entry is created, a new device id is generated anyway. I've just taken it out of the form, but it is still used in the config entry data, just a new one is always generated.

Reply:
There are ways to run code when an entry was deleted, so if you could also deregister we could optimize this and avoid the problem altogether. (Like it could be that you didn't know that that was a thing).

But this is also fine 👍🏻